daffodil-core/src/main/scala/org/apache/daffodil/core/dpath/Expression.scala

```diff
@@ -1510,7 +1510,7 @@ case class FunctionCallExpression(functionQNameString: String, expressions: List
           functionQNameString,
           functionQName,
           NodeInfo.String,
-          FNLocalName0
+          () => FNLocalName0()
```

Comment:
I'm a little confused by the syntax on these.  Is this parameter a function that returns an FNLocalName0 function?  Can we no longer just pass in FNLocalName0?

Reply:
I'm guessing there's an ambiguity if `FNLocalName0` is a lambda or if the parenthesis should be auto-applied so Scala now creates a warning? Seems it like should be obvious passes on the FNZeroArgExpr argument type, but maybe not.

And I'm not sure there's a way to use the `_` syntax to obviously make it a lambda like we can with the FNFoo functions that accept parameters. But in my testing, it looks like passing in `FNLocalName0.apply` seems to work and doesn't create a warning, and makes it so the `lazy val constructor` isn't needed.

Reply:
Steve is right, it gives an auto application error with the naked FNLocalName0...and I think it gave an additional error even with the FNLocalName0.apply...something about functions not being able to be passed as call by name...let me undo the changes and confirm

Reply:
Ah `.apply` worked, but I was doing `.apply()`, which it did not like. Will fix
```
[error] -- [E007] Type Mismatch Error: ...daffodil-core/src/main/scala/org/apache/daffodil/core/dpath/Expression.scala:1528:31 
[error] 1528 |          FNNamespaceUri0.apply()
[error]      |          ^^^^^^^^^^^^^^^^^^^^^^^
[error]      |        Found:    org.apache.daffodil.runtime1.dpath.FNNamespaceUri0
[error]      |        Required: () => org.apache.daffodil.runtime1.dpath.RecipeOp
```